### PR TITLE
Support recovering Nokia devices via console

### PR DESCRIPTION
**Pull request by yutongzhang-microsoft**
### Description of PR

In PR #10806, we support auto recovery of testbeds via console. But at that time, we didn't support nokia testbeds. 
In Nokia testbeds, they first enter into `u-boot`, and we can enter into `ONIE` through `u-boot`. So in this PR, we enhance the function which will do recover in ONIE to cover this sistuation. 

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305

#### What is the motivation for this PR?
In PR #10806, we support auto recovery of testbeds via console. But at that time, we didn't support nokia testbeds. 
In Nokia testbeds, they first enter into `u-boot`, and we can enter into `ONIE` through `u-boot`. So in this PR, we enhance the function which will do recover in ONIE to cover this sistuation. 

#### How did you do it?
Enhance the function which will do recover in ONIE to cover Nokia testbeds. If the type of testbeds is Nokia, enter into ONIE through u-boot, and then, recover in ONIE. 

#### How did you verify/test it?
Recover Nokia testbeds.

### CODE REVIEW

File: .azure-pipelines/recover_testbed/common.py [hunk 1]
```diff
@@ -71,6 +71,14 @@ def posix_shell_onie(dut_console, mgmt_ip, image_url, is_nexus=False):
                         dut_console.remote_conn.send('reboot\n')
                         continue
 
+                    if is_nokia and enter_onie_flag is True:
```
Comment (yejianquan) [inline .azure-pipelines/recover_testbed/common.py:74]: Have you figured out whether the sequence and priority affect boolean usage?

File: .azure-pipelines/recover_testbed/common.py [hunk 2]
```diff
@@ -71,6 +71,14 @@ def posix_shell_onie(dut_console, mgmt_ip, image_url, is_nexus=False):
...
+                        if "stop autoboot" in x:
```
Comment (yejianquan) [inline .azure-pipelines/recover_testbed/common.py:75]: Please create constants for magic strings

File: .azure-pipelines/recover_testbed/common.py (hunk 1 above)
Comment (yutongzhang-microsoft) [reply .azure-pipelines/recover_testbed/common.py:74]: This sequence will work as expected

**Review by yejianquan — APPROVED** [review]
LGTM

**Review by wangxin — APPROVED** [review]

**Pull request merged**